File: app/src/main/java/com/fintamath/textview/MathAlternativesTextView.java
package com.fintamath.textview;

import android.content.Context;
import android.content.res.TypedArray;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.Nullable;

import com.fintamath.R;

import java.util.List;

public class MathAlternativesTextView extends LinearLayout {

    private final int mTextViewLayout;
    private final int mDelimiterLayout;
    private final int mLayout;

    private final LayoutInflater mInflate;
    private final TextView mMainTextView;

    public MathAlternativesTextView(Context context) {
        this(context, null);
    }

    public MathAlternativesTextView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);

        setOrientation(VERTICAL);

        TypedArray a = context.obtainStyledAttributes(attrs, R.styleable.MathAlternativesTextView);

        mLayout = a.getResourceId(R.styleable.MathAlternativesTextView_alternativeLayout, 0);
        mTextViewLayout = a.getResourceId(R.styleable.MathAlternativesTextView_alternativeTextViewLayout, 0);
        mDelimiterLayout = a.getResourceId(R.styleable.MathAlternativesTextView_alternativeDelimiterLayout, 0);

        a.recycle();

        mInflate = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);

        mMainTextView = (TextView) mInflate.inflate(mTextViewLayout, null);
        addTextView(mMainTextView);
    }

    public void setTexts(@Nullable List<String> texts) {
        removeViews(1, getChildCount() - 1);

        if (texts == null || texts.isEmpty()) {
            mMainTextView.setText("");
            return;
        }

        mMainTextView.setText(texts.get(0));

        for (int i = 1; i < texts.size(); i++) {
            if (!isTextUnique(texts.get(i))) {
                continue;
            }

            TextView alternativeTextView = (TextView) mInflate.inflate(mTextViewLayout, null);
            addView(mInflate.inflate(mDelimiterLayout, null));
            addTextView(alternativeTextView);
            alternativeTextView.setText(texts.get(i));
        }
    }

    private void addTextView(TextView textView) {
        ViewGroup scrollView = (ViewGroup) mInflate.inflate(mLayout, null);
        scrollView.setForegroundGravity(getForegroundGravity());
        scrollView.addView(textView);
        addView(scrollView);
    }

    private boolean isTextUnique(String text) {
        for (int i = 0; i < getChildCount(); i++) {
            View view = getChildAt(i);
            if (view instanceof ViewGroup) {
                ViewGroup viewGroup = (ViewGroup) view;
                TextView textView = (TextView) viewGroup.getChildAt(0);
                if (textView.getText().toString().equals(text)) {
                    return false;
                }
            }
        }

        return true;
    }
}

File: app/src/main/java/com/fintamath/textview/MathAlternativesTextView.kt
package com.fintamath.textview

import android.annotation.SuppressLint
import android.content.Context
import kotlin.jvm.JvmOverloads
import android.widget.LinearLayout
import android.view.LayoutInflater
import android.util.AttributeSet
import android.view.View
import com.fintamath.R

class MathAlternativesTextView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null
) : LinearLayout(context, attrs) {

    private val mathTextViewLayout: Int
    private val delimiterLayout: Int

    private var onTouchListener: OnTouchListener? = null

    private val inflate: LayoutInflater
    private val textViews = mutableListOf<MathTextView>()
    private val delimiters = mutableListOf<View>()

    init {
        orientation = VERTICAL

        val a = context.obtainStyledAttributes(attrs, R.styleable.MathAlternativesTextView)

        mathTextViewLayout =
            a.getResourceId(R.styleable.MathAlternativesTextView_alternativeTextViewLayout, 0)
        delimiterLayout =
            a.getResourceId(R.styleable.MathAlternativesTextView_alternativeDelimiterLayout, 0)

        a.recycle()

        inflate = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater

        addTextView()
    }

    fun setTexts(texts: List<String>) {
        for (i in textViews.size until texts.size) {
            addDelimiter()
            addTextView()
        }

        textViews[0].text = texts[0]

        val distinctTexts = texts.distinct()

        for (i in 1 until distinctTexts.size) {
            textViews[i].text = distinctTexts[i]
            textViews[i].visibility = VISIBLE
            delimiters[i - 1].visibility = VISIBLE
        }

        for (i in distinctTexts.size until textViews.size) {
            textViews[i].visibility = GONE
            textViews[i].clear()
            delimiters[i - 1].visibility = GONE
        }
    }

    @SuppressLint("ClickableViewAccessibility")
    override fun setOnTouchListener(listener: OnTouchListener) {
        onTouchListener = listener

        for (text in textViews) {
            text.setOnTouchListener(onTouchListener)
        }

        super.setOnTouchListener(onTouchListener)
    }

    @SuppressLint("ClickableViewAccessibility")
    private fun addTextView() {
        val textView = inflate.inflate(mathTextViewLayout, null) as MathTextView
        textView.setOnTouchListener(onTouchListener)
        textViews.add(textView)
        addView(textView)
    }

    private fun addDelimiter() {
        val delimiter = inflate.inflate(delimiterLayout, null)
        delimiter.visibility = GONE
        delimiter.setOnTouchListener(onTouchListener)
        delimiters.add(delimiter)
        addView(delimiter)
    }
}
